app/app.js

```diff
@@ -92,7 +95,7 @@ const render = (messages) => {
 if (module.hot) {
   // modules.hot.accept does not accept dynamic dependencies,
   // have to be constants at compile-time
-  module.hot.accept('./i18n', () => {
+  module.hot.accept(['./i18n', 'containers/App'], () => {
```

Comment:
As I recall it depends on whether you import the component with `require`/`import` initially, and I think whether it is the default or a named export also has an effect. I couldn't find the original info on that, but the current [example in `react-hot-boilerplate`](https://github.com/gaearon/react-hot-boilerplate/blob/next/src/index.js) also does it without an additional require.